### DISCUSSION TITLE
Extensions Unloading on VSC Reload

### DISCUSSION THREAD

**Discussion by scottfurry** [microsoft/vscode-discussions#386] [Extension Development QnA] [answered]
Extensions affected:
-[Shellcheck](https://marketplace.visualstudio.com/items?itemName=timonwong.shellcheck) ver 29.3 (updated 2023-01-04)
-[Material Icon Theme](https://marketplace.visualstudio.com/items?itemName=PKief.material-icon-theme) ver 4.23.1 (updated 2022-12-23)

In both instances, extensions appear to be installed successfully. On a reboot, functionality provided by extensions starts up and then disappears. Environment then reverts to any default settings given loss of extension (e.g. Icon theme reverts to minimal or seti). Extensions no longer appear in listing (both enabled or disabled). These extensions have been "unloaded" or uninstalled somehow during environment startup.

Re-install after clearing cache does not resolve situation. Only workaround is to install previous revision of extension. 

How to debug extensions to give extension maintainer meaningful information?

**Comment by Austindgk232** [comment]
> Extensions affected: -[Shellcheck](https://marketplace.visualstudio.com/items?itemName=timonwong.shellcheck) ver 29.3 (updated 2023-01-04) -[Material Icon Theme](https://marketplace.visualstudio.com/items?itemName=PKief.material-icon-theme) ver 4.23.1 (updated 2022-12-23)
> 
> In both instances, extensions appear to be installed successfully. On a reboot, functionality provided by extensions starts up and then disappears. Environment then reverts to any default settings given loss of extension (e.g. Icon theme reverts to minimal or seti). Extensions no longer appear in listing (both enabled or disabled). These extensions have been "unloaded" or uninstalled somehow during environment startup.
> 
> Re-install after clearing cache does not resolve situation. Only workaround is to install previous revision of extension.
> 
> How to debug extensions to give extension maintainer meaningful information?

**Comment by ArturoDent** [comment, marked as answer]
Can you try the Insiders Build?  There have been a couple of similar issues that have been worked on since the last release.

**Reply by scottfurry** [reply]
There was some change _upstream_ (exactly where is a mystery). After a downgrade of extensions, restart, and re-install _something_ changed and the extensions stopped causing issues. My running theory is a change to extension structure was being promulgated but not exactly implemented elsewhere.

tl;dr - Problem resolved due to unknown changes
